modules/lang-painless/src/main/java/org/opensearch/painless/action/PainlessExecuteAction.java

```diff
@@ -182,7 +182,7 @@ static class ContextSetup implements Writeable, ToXContentObject {
 
             static ContextSetup parse(XContentParser parser, Void context) throws IOException {
                 ContextSetup contextSetup = PARSER.parse(parser, null);
-                contextSetup.setXContentType(parser.contentType());
+                contextSetup.setXContentType((XContentType) parser.contentType());
```

Comment:
These typecases could bite as in the future, I am wondering should we add the method to `XContentType`:

```
    public static XContentType fromMediaType(MediaType mediaType) {
        if (mediaType instanceof XContentType) {
            return (XContentType) mediaType;
        } else {
            return MEDIA_TYPE_PARSER.fromMediaType(mediaType.mediaTypeWithoutParameters());
        }
    }
```

In this case all typecast will be isolated.

Reply:
Yes! I like that idea. Done.